### DISCUSSION TITLE
Give us back ILAG guest login, please!

### DISCUSSION THREAD

**Discussion by r4dh4l** [element-hq/element-meta#728] [Ideas]
### Description

I already mentioned this in https://github.com/vector-im/riot-web/issues/8808#issuecomment-465797492 but maybe the issue topic is not emphasizing my point:

Before Riot 1.0 it was possible to join a room _without_ creating an account. Yes, choosing a user name was actually technically creating an account on the according home server but in perspective of the user it was a way to join a conversation without the registration procedure.
This feature is very essential in my opinion because you could invite users on project pages to a project room without bothering them to create an account in the first place.

### Steps to reproduce

1. Join a room with allows guest access via Riot web app.
2. Press "Click here to join the discussion!" and you will get an info box for "Registration Required". Before Riot 1.0 you could just choose a username and join the room.

> Describe how what happens differs from what you expected.

After "Click here to join the discussion!" I would like to see a button "Join as guest" next to "Cancel", "Register" (and maybe "Join with existing Account").
Edit:
After "Join as guest" there should be an input field with pre-generated username like `@guest_expiring2019-10-12_13-42-46_<CustomPart>:matrix.org` while only `<CustomPart>` can be edited. This would mean:

* that the guest account expiration is 24h (this could be a home server setting/default)
* the a guest account was created on 2019-10-11 13:42:46
* the guest account will expire on 2019-10-12 13:42:46.

This way a guest account would _clearly_ indicate that

* it is a guest account
* it is not possible to contact such an account after 2019-10-12 13:42:46

- **Platform**: web

For the web app:

- **Browser**: Firefox
- **OS**: GNU/Linux Debian
- **URL**: riot.im/app

**Comment by t3chguy** [comment]
The issue is creating an account is irreversible, so the usernames become burned when someone stops using the "guest" account as matrix doesn't support account deletion for various privacy/access permissions reasons.
A lot of people complained they logged out and didn't set an email so can no longer use their usernames

**Comment by ara4n** [comment]
hang on though - Riot 1.0 hasn't deliberately removed any guest access.  The change @t3chguy describes happened ages ago (Riot 0.11 or something).

You should still be able to view a guest-accessible room without having to create an account.

**Comment by t3chguy** [comment]
Didn't ILAG get disabled for 1.0? The magical thing which created an account with password in the background with a username of your choosing

**Comment by ara4n** [comment]
oh, i see - yes, ILAG got disabled, mainly because we didn't have time to adopt it for the new login flows.  So the problem isn't that guest login is missing (it's there as much as ever), but registering-real-accounts-for-guests-in-the-background is missing.

**Comment by r4dh4l** [comment]
thx @t3chguy for your reply.

> The issue is creating an account is irreversible, so the usernames become burned when someone stops using the "guest" account as matrix doesn't support account deletion for various privacy/access permissions reasons.

I understand but a pre-generated guest name like `guest885e24fa1c` wouldn't "burn" anything someone would like to use later AND indicate that the currently used account is a guest account. ;)

> A lot of people complained they logged out and didn't set an email so can no longer use their usernames

Well, a "join as guest" button with an according warning would be enough to avoid complaining users (at least those who are willing to read and think before clicking a button).

Thx @ara4n for your reply as well.

> So the problem isn't that guest login is missing (it's there as much as ever), but registering-real-accounts-for-guests-in-the-background is missing.

So should I rename the issue title to "Registering real accounts for guests in the background is missing"?

Edit: Just noticed the title was already renamed. Don't know what "ILAG" means but it seems okay for you devs. ;)

**Comment by t3chguy** [comment]
But then he issue is that an account with a username like guestXXXXX is undesirable and account migration is not a feature of the protocol at this time so users might end up stuck with a crappy username

**Comment by r4dh4l** [comment]
There is no need for an account migration in my opinion.

Related to the case of "undesirable usernames": Let's take a look on the possibilties a user without a Matrix ID has when invited to a room and has the choice between

- "Cancel"
- "Register"
- "Join with existing account"
- "Join as guest"

Let's assume the user chooses "join as guest" and gets the username `guest885e24fa1c` with an explicit info like

> Guest accounts are intended to provide users room access for public rooms without forcing them to register an account and for just a single session. If you want to use a more personalized account and want to participate in long-term you should _register_ a standard account (giving you the possibility to choose a user name of your choice).

If the user is not happy with the username `guest885e24fa1c` the user would obviously choose to go back and register a personal account.

At the moment the situation is like this: Because Matrix offered to access rooms as guest I promoted this feature on quite a lot of pages with "feel free to join our matrix channel (guest account/without registration possible)". Now theses users end up on a page forcing them to register an account. The "I don't care where I create an account"-faction won't have a problem to do this, but the "we actually don't like to create an account just to participate in a short conversation"-faction will not (the latter is the one I try to address with a federated, free messenger system like Matrix).

**Comment by jtagcat** [comment]
Merged issue from vector-im/element-web#9316 I'd suggest to Change the title to just "Better signup flow" @r4dh4l 
## Issue
Currently, when you open a link like [this](https://riot.im/app/#/user/@c7c7:matrix.org?action=chat), you are required to sign up. With Matrix.org/Riot.im being so small, it is a real obstacle on getting new users to even try it.
I direct people at my riot link, they come back and say:
 - It required an a account
 - Wasn't fast enough
 - ew login page, I'm not giving my details away
Remember, new people probably do not know, that Riot.im can let you Communicate the way you want - a universal secure chat app entirely under your control and just want to chat.
Also the Register button as seen below unconsciously means give me your email and password please, to people who don't know riot.

![image](https://user-images.githubusercontent.com/38327267/55188241-ca24df80-5192-11e9-92be-9d9cc4df7767.png)

### Current flow:
1) The page loads for 10s long, you need to press register
2) after 5-10s Register form loads, oh no they need my details! I think of an username for 10s, gonna reuse my unsecure password! +5s
3) Captcha (on chrome +5-20s, on firefox +15-50s). If you decide to enter an email you get punished by also needing to confirm the email (let an account be for few days, without confirming email please) +60s notice: if you use email you will also not get the start chatting button in the tab the email opened in! or if you choose to not use email, you will get a prompt to use an email +5s for click
4) You finally solve the captcha and maybe had to go to your inbox, the page loads for 5sec, you get a sound notification, the riot-bot wants your attention! (stop it for users registered via a link to chat with somebody), you get the start chatting button, you press it, you wait 10sec and chat.

The flow is pretty bad and has many clicks and annoyances. I did it faster at 2 minutes, a normal user will take even more. At this point you have probably forgotten, why are you even in riot, especially when opening my link and using email verification the start chat disappears to the original tab. (note in the gif I use keyboard shortcut to switch back to the original tab, to show the start chat, peek didn't capture my tabs)
I have tried not comparing to Di***rd, but there you are in with 10s + choosing an username 10s The faster, the better, I'd call below 40s good for Riot. 

[video, converted for faster viewing](https://youtu.be/oBit83OUhAk)

# Instead:
After clicking a link to initiate chat, automatically create the user a temporary account, what can be only be logged on to with cookies, the account expires say after a week:
  a) have the account on matrix.org
  b) if it is an issue for the public servers, make this feature exclusive to custom domain users, for example if my account is @jtagcat:domain.tld, clicking my link would make you an account @temp569:domain.tld or have the user choose **only** an username.
The account should be upgradable to a real account (to convert temp users to real users).
A new user does not have many rooms or people, I suggest using the dark purple area for a small introduction of riot and an enter password (and email) field.
If we can copy the few sent messages from @temp569:matrix.org (or change the username and/or server), give them the normal sign up form right there. It is important that the form is not hidden behind a link.

![image](https://user-images.githubusercontent.com/38327267/55188800-14f32700-5194-11e9-8883-693da7cb75c4.png)

If I am unclear, please ask! I hope the flow gets fixed, tough I don't know any JS (yet) in order to help, I contribute via opencollective.

**Comment by tgr** [comment]
This would be very useful for using Riot as an IRC webchat interface. E.g. for Freenode people are currently using [webchat.freenode.net](https://webchat.freenode.net/), which is rather ugly and does not show past messages (so if you join, ask a question and leave, there is no way to see the answers you have got), Riot would be a great replacement for that, except casual users will abandon it rather than register.

**Comment by jtagcat** [comment]
@tgr I would just point out that https://kiwiirc.com/ is a good option for irc (though it doesn't hold the history).

**Comment by r4dh4l** [comment]
Watching the label of this issue changing from "fire" to "<SomethingIdontRemember>" to "bug": May I ask what is the problem to enable guest login again?

**Comment by t3chguy** [comment]
Guests led to a bad long-term experience, given there is no proper upgrade path, so if you start off on a guest account you need to register a proper account later on if you want to be able to access it long-term and then you need to get a new username, need to rejoin rooms etc

**Comment by r4dh4l** [comment]
> Guests led to a bad long-term experience, given there is no proper upgrade path, so if you start off on a guest account you need to register a proper account later on if you want to be able to access it long-term and then you need to get a new username, need to rejoin rooms etc

Okay but the reason for this is just that the UI didn't explain well what guest accounts are about. What about my initial idea

> After "Join as guest" there should be an input field with pre-generated username like `tempuser<RandomGenerated40bitHexKey>_CustomPart` (40bit Hex-Key or something) allowing the guest to adjust `CustomPart` (this would prevent guests to choose probably already existing user names).
The guest login process should end up with an info box that guest accounts will be deleted after X hours (defined by the home server settings).

?

Currently the "bad long-term experience" is that I've promoted [matrix] as communication network you don't need to register an account if you don't want to which made it very easy to invite external people to conversations (for an interview etc). A whole use case of [matrix] suddenly died by disabling the guest login.

**Comment by t3chguy** [comment]
Guests are notated by their usernames being just numbers, so that suggestion doesn't work. As for deletion of accounts, it can only cause confusion, what if I learn that my friend is @13214:matrix.org then X hours later it could be someone else and I may have told them some private information accidentally

**Comment by t3chguy** [comment]
> A whole use case of [matrix] suddenly died by disabling the guest login.

[matrix] still supports it, just modern versions of riot-web does not, there are other clients which do/can support it.
